metricbeat/module/kubernetes/state_container/state_container.go

```diff
@@ -106,15 +102,17 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	}, nil
 }
 
-// Fetch methods implements the data gathering and data conversion to the right format
-// It returns the event which is then forward to the output. In case of an error, a
-// descriptive error must be returned.
-func (m *MetricSet) Fetch() ([]common.MapStr, error) {
+// Fetch methods implements the data gathering and data conversion to the right
+// format. It publishes the event which is then forwarded to the output. In case
+// of an error set the Error field of mb.Event or simply call report.Error().
+func (m *MetricSet) Fetch(reporter mb.ReporterV2) {
 	m.enricher.Start()
 
 	events, err := m.prometheus.GetProcessedMetrics(mapping)
 	if err != nil {
-		return nil, err
+		m.Logger().Error(err)
```

Comment:
You directly want to use the new interface?

Reply:
Do you mean to use `Fetch(reporter mb.ReporterV2) error`?? `ptest` package needs to be updated too to use it... and it might affect prometheus module too, that's why I haven't used it

Reply:
yes. In that case, let's delay it.